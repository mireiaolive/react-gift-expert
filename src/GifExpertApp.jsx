import React from "react";
import { useState } from "react";
import { AddCategory, GifGrid } from "./components";

//para actualizar el state usamos el setCategories
//con un nuevo arreglo
export const GifExpertApp = () => {
    const [categories, setCategories] = useState(["Batman"]);

    const onAddCategory = (newCategory) => {
        //si la nueva categoria ya existe no la incluyas
        if (categories.includes(newCategory)) return;
        // console.log(newCategory);
        // categories.push(newCategory);
        setCategories([newCategory, ...categories]);
    };
    return (
        <>
            <h1>GifExpertApp</h1>
            {/* <AddCategory setCategories={setCategories} /> */}
            {<AddCategory onNewCategory={(event) => onAddCategory(event)} />}

            <ol>
                {categories.map((category) => (
                    <GifGrid key={category} category={category} />
                ))}
            </ol>
        </>
    );
};
