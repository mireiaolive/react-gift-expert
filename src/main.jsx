import ReactDOM from "react-dom/client";
// import { FormWithCustomHook } from "./02-useEffect/FormWithCustomHook";
// import { SimpleForm } from "./02-useEffect/SimpleForm";
// import { CounterWithCustomHook } from "./01-useState/CounterWithCustomHook";
import "./index.css";
import { MultipleCustomHooks } from "./03-examples/MultipleCustomHooks";

ReactDOM.createRoot(document.getElementById("root")).render(
    // <React.StrictMode>
    <MultipleCustomHooks />
    // </React.StrictMode>
);
