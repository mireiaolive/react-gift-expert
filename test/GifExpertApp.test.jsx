import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import { GifExpertApp } from "../src/GifExpertApp";

describe("Pruebas en <GifexpertApp/>", () => {
    test("debe hacer match con el snapshot", () => {
        const { container } = render(<GifExpertApp />);
        expect(container).toMatchSnapshot();
        //screen.debug();
    });

    test("Debe de mostrar el titulo en el componente", () => {
        render(<GifExpertApp />);
        expect(screen.getByText("GifExpertApp"));
    });

    test("No debe agregar una categorÃ­a si ya existe", () => {
        render(<GifExpertApp />);

        const input = screen.getByPlaceholderText("search gifs");
        const form = screen.getByRole("form", { name: "form" });

        fireEvent.change(input, { target: { value: "Dragon Ball" } });
        fireEvent.submit(form);

        const categories = screen.getAllByText("Dragon Ball");
        expect(categories.length).toBe(1);
    });
});
