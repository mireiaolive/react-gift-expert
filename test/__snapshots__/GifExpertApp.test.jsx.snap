// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifexpertApp/> debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="search gifs"
      type="text"
      value=" "
    />
  </form>
  <ol>
    <h3>
      Batman
    </h3>
    <h2>
      Cargando...
    </h2>
    <div
      class="card-grid"
    />
  </ol>
</div>
`;
